### ISSUE TITLE
[Feature]:  Support cast decimal as bigint

### ISSUE THREAD

**Issue by domingozhang** [matrixorigin/matrixone#3249]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

select cast(-19999999999999999999 as signed);
Operator 'unary_minus' with parameters [VARCHAR] will be implemented in future version.

**Comment by domingozhang** [comment]
mysql> select cast(-19999999999999999999 as signed);
ERROR 1105 (HY000): Operator 'cast' with parameters [DECIMAL128 BIGINT] will be implemented in future version.

@m-schen bigint to decimal128 seems not supported

**Comment by m-schen** [comment]
> mysql> select cast(-19999999999999999999 as signed); ERROR 1105 (HY000): Operator 'cast' with parameters [DECIMAL128 BIGINT] will be implemented in future version.
> 
> @m-schen bigint to decimal128 seems not supported

we heve bigint -> decimal128 now.
but not support decimal128 -> bigint at v0.5

**Comment by m-schen** [comment]
Track it with a new issue tag.
we do not have decimal128 -> bigint now.
but I'm no sure if we will fix it at version 0.6 @iamlinjunhong

**Comment by iamlinjunhong** [comment]
Doesn't support cast decimal as bigint now.  Discuss with @domingozhang that will put this issue to 0.6.
